scripts/src/bold/id_engine.py

```diff
@@ -170,38 +170,55 @@ def _bold_sequence_search(
                         float(longitude_str) if longitude_str else None),
                 }
                 sequence_hits.append(result)
+            return sequence.id, sequence_hits
 
-            # If multiple seq IDs, only keep the one with hits
-            # (the correct orientation)
-            if (
-                sequence.id not in hits
-                or sequence_hits
-            ):
-                hits[sequence.id] = {
-                    'query_id': sequence.id,
-                    'query_title': sequence.description,
-                    'query_length': len(sequence.seq),
-                    'query_frame': sequence.annotations.get("frame"),
-                    'query_strand': (
-                        '+'
-                        if sequence.annotations.get("forward", True)
-                        else '-'
-                    ),
-                    'query_sequence': str(sequence.seq),
-                    'query_orientation': (
-                        'HMMSearch'
-                        if sequence.annotations.get("oriented")
-                        else 'BOLD ID Engine'
-                    ),
-                    'hits': sequence_hits,
-                }
+        # If multiple seq IDs, only keep the one with hits
+        # (the correct orientation)
+        # Submit all sequences concurrently
+        with ThreadPoolExecutor(max_workers=50) as executor:
+            future_to_seq = {
+                executor.submit(submit_sequence, sequence, i): sequence
+                for i, sequence in enumerate(sequences)
+            }
 
-        ordered_hits = {
-            seq.id: hits[seq.id]
-            for seq in sequences
-        }
+            # Collect results as they complete
+            for future in as_completed(future_to_seq):
+                sequence_id, sequence_hits = future.result()
+                if (
+                    sequence_id not in hits
+                    or sequence_hits
+                ):
+                    hits[sequence_id] = {
+                        'query_id': sequence_id,
+                        'query_title': future_to_seq[future].description,
```

Comment:
This would be a bit more clear if you just do `sequence = future_to_seq[future]` and refer to `sequence` throughout?